reflectionserializer/README.md

```diff
@@ -0,0 +1,248 @@
+* optionally, slices can have duplicates removed before serialization and deserialization;
+* by default, interface and pointer types cannot have `nil` value;
+* by default, sizes of dynamic containers (`string`, `slice`) are written before values as uint32;
+* optionally, collection types' length (e.g. `slice`, `orderedmap.OrderedMap`) can be validated before serialization and
+  deserialization;
+* golang's native `map` type is not supported;
+* structs are serialized in the order of fields in the struct;
+* only public fields can be serialized. To use private fields, read more about `unpack` struct tag;
+* `encoding.BinaryUnmarshaler` and `encoding.BinaryMarshaler` interfaces are supported and used if available;
+    * serialized bytes are serialized as a regular byte slice;
+    * during deserialization byte slice is read and passed to unmarshaler;
+* for custom collection types (e.g. `orderedmap.OrderedMap`), a special `BinarySerializer` and `BinaryDeserializer` is
+  used. Custom collection type needs to implement those interfaces internally;
+
+## Available struct tags
+
+Structure fields can be annotated using struct tags that can specify the following serialization details:
+
+| Struct tag name | Type | Description                                                                                        | Supported types                          |
+|-----------------|------|----------------------------------------------------------------------------------------------------|------------------------------------------|
+| serialize       | bool | only fields with this tag will be serialized                                                       | all                                      |
+| unpack          | bool | unpacks the inner structure's fields as fields of outer structure                                  | `struct`                                 |
+| minLen          | int  | ensure that serialized collection has the specified min length.                                    | `slice` `orderedmap.OrderedMap`          |
+| maxLen          | int  | ensure that serialized collection has the specified max length                                     | `slice` `orderedmap.OrderedMap`          |
+| allowNil        | bool | whether given struct field can have nil value                                                      | `interface` `ptr`                        |
+| noDuplicates    | bool | whether a slice should have duplicate values removed during serialization and deserialization      | `slice`                                  |
+| lexicalOrder    | bool | whether a slice values should be lexicographically sorted before serialization and deserialization | `slice`                                  |
+| lenPrefixBytes  | int  | number of bytes to use for serialization of length of types with dynamic size.                     | `slice` `orderedmap.OrderedMap` `string` |
```

Comment:
This should instead enforce the rules instead of auto. doing it; add specific tags for auto. removal/ordering.

Reply:
Added two tags - `sort` and `skipDuplicates`. Previous tags return an error if order/duplicates requirement is violated.